make/dev.mk

```diff
@@ -13,27 +13,49 @@ TIMESTAMP:=$(shell date +%s)
 TAG?=$(GIT_COMMIT_ID_SHORT)-$(TIMESTAMP)
 OPENSHIFT_VERSION?=4
 
-.PHONY: create-resources
-create-resources:
-	$(info Logging using system:admin...)
-	$(Q)oc login -u system:admin
-	$(info Creating sub resources...)
-	$(info Creating CRDs...)
-	$(Q)oc create -f ./deploy/crds/devconsole_v1alpha1_gitsource_crd.yaml
-	$(info Creating Namespace)
-	$(Q)oc create -f ./deploy/namespace.yaml
-	$(info oc project codeready-devconsole)
-	$(Q)oc project codeready-devconsole
-	$(info Creating Service Account)
-	$(Q)oc create -f ./deploy/service_account.yaml
-	$(info Creating Role)
-	$(Q)oc create -f ./deploy/role.yaml
-	$(info Creating RoleBinding)
-	$(Q)oc create -f ./deploy/role_binding.yaml
-
-.PHONY: create-cr
-create-cr:
-	$(info Creating Custom Resource...)
+# to watch all namespaces, keep namespace empty
+APP_NAMESPACE ?= ""
+LOCAL_TEST_NAMESPACE ?= "local-test"
+
+.PHONY: local
+## Run Operator locally
+local: deploy-rbac build deploy-crd
+	$(Q)-oc new-project $(LOCAL_TEST_NAMESPACE)
+	$(Q)operator-sdk up local --namespace=$(APP_NAMESPACE)
+
+.PHONY: deploy-rbac
+## Setup service account and deploy RBAC
+deploy-rbac:
+	$(Q)-oc login -u system:admin
+	$(Q)-oc create -f deploy/service_account.yaml
+	$(Q)-oc create -f deploy/role.yaml
+	$(Q)-oc create -f deploy/role_binding.yaml
+
+.PHONY: deploy-crd
+## Deploy CRD
+deploy-crd:
+	$(Q)-oc apply -f deploy/crds/devconsole_v1alpha1_component_crd.yaml
+	$(Q)-oc apply -f deploy/crds/devconsole_v1alpha1_gitsource_crd.yaml
+	$(Q)-oc apply -f deploy/crds/devconsole_v1alpha1_gitsourceanalysis_crd.yaml
+
+.PHONY: deploy-operator
+## Deploy Operator
+deploy-operator: deploy-crd
+	$(Q)oc create -f deploy/operator.yaml
+
+.PHONY: deploy-clean
+## Deploy a CR as test
+deploy-clean:
+	$(Q)-oc delete project $(LOCAL_TEST_NAMESPACE)
+
+.PHONY: deploy-test
+## Deploy a CR as test
+deploy-test:
+deploy-test:
```

Comment:
@corinnekrych can we remove the duplicate "deploy-test:" line?

Reply:
@tinakurian done in c2e15ad